lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -279,35 +301,58 @@ export function usePolledAPIFetch<T = unknown>({
     // When the actual request is loading, transition to loading, in case
     // a new request was triggered by a path/fetch key change
     if (result.isLoading) {
+      // If the fetch is loading but not due to a refresh, it means we are
+      // fetching from a new source. In that case reset the refresh count and
```

Comment:
```suggestion
      // fetching a new URL. In that case reset the refresh count and
```

"From a new source" seemed unclear to me. What we mean is that the path or parameters have changed.